MITADS/opensubtitles_exporter.py

```diff
@@ -0,0 +1,354 @@
+        year_folder_int = int(year_folder)
+        if(year_folder_int<1300 or year_folder_int>=start_year):
+          xml_file_list.append(str(path))
+      except:
+        pass
+
+  
+  total_files = len(xml_file_list)
+  print('total xml file: '+ str(total_files))
+
+  file_output = None
+  if(len(xml_file_list)>0):    
+    file_output=open("opensubtiles_extracted.txt","w")
+
+  count_file = 0
+  count_sentence = 0
+  count_split_file = 1
+  for xml_file_path in xml_file_list:
+    try:
+      fp = open(xml_file_path,encoding='utf-8')
+      
+      prev_time_line = None
+      exported_file = False
+      for line in fp:
+        _line= line
+
+        #if('$' in _line):
+        #  deb = 0
+
+        line=line.strip()
+        if(line=='' or len(line)<2):
+          continue
+
+        ##le linee con il testo delle frasi non iniziano con apertura/chiusura tag  
+        if(line[0]!='<'):
+          ##clear and normalize
+          line =  process_text(line)
+          ##############
+          if(line=='' or len(line)<2):
+            continue
+
+          ##debug check - abilita se vuoi avere un output con le segnalazioni delle righe che richiedono attenzione o ulteriore pulizia
+          if(False):
+            regexp_apex_ok = re.compile(r'\w\'\w')
+            line_temp = line.replace(" ", "")
+            if(not line_temp.isalnum() 
+                and '!' not in line_temp 
+                and '?' not in line_temp
+                and '.' not in line_temp
+                and ':' not in line_temp
+                and ',' not in line_temp
+                and ';' not in line_temp
+                #and '"' not in line_temp 
+                and not regexp_apex_ok.search(line_temp) ):
+              print('IMPROVE TEXT CLEANING: ' + line + ' - source: '+_line) 
+          #####################################################
+
+          ##splitta il risultato in più file, se abilitato il flag split_output_file_rows, con il numero di righe per file
+          if split_output_file_rows!=None and not (count_sentence % split_output_file_rows):
+            count_sentence = 0
+            file_output.close()
+            file_output=open("opensubtiles_extracted_{}.txt".format(str(count_split_file)),"w")
+            count_split_file +=1
+          ####
+          try:
+            file_output.write(line+ '\n')
+            count_sentence +=1
+          except Exception as e:
+            ###QUESTO CODICE SI PUO' ELIMINARE _ ADESSO NON VA PIU' IN ERRORE
+            
+            print('ERROR WRITE ROW - '+str(e) + ' - processed: ' +line + ' - source: '+_line)
+            
+            ##remove all non-ascii char
+            line = re.sub(r'[^\x00-\x7F]+','', line)
+            file_output.write(line+ '\n')
+            count_sentence +=1
+            #raise(e)
+          
+          exported_file = True
+
+        ##prev_line to check delay time, todo.. 
+        #if(line.startswith('<time')):          
+        #  prev_line = line
```

Comment:
We need that?

Reply:
If it is important to separate sentences based on waiting breaks,, decomment code and continue to dev...

Reply:
so we can implementing with a parameter to the script and remove other commented code

Reply:
sorry I did not understand the parameter for what?
The code that takes the time-pause between subtitles has yet to be implemented. 
It was just a sketch, maybe it's better if I delete it then ...

Reply:
....... then must determine when a pause determines a new conversation block...

Reply:
I mean a parameter to the script itself :-)